extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -175,12 +188,37 @@ export class AmpStoryReaction extends AMP.BaseElement {
   }
 
   /**
-   * Generates the template in rootEl_ and fills up with options.
-   * @param {!Element} unusedElement
+   * Reads the element attributes prefixed with option- and returns them as a list.
+   * eg: [
+   *      {optionIndex: 0, text: 'Koala'},
+   *      {optionIndex: 1, text: 'Developers', correct: 'correct'}
+   *    ]
+   * @protected
+   * @return {!Array<!OptionConfigType>}
+   */
+  parseOptions_() {
+    const options = [];
+    toArray(this.element.attributes).forEach((attr) => {
+      // Match 'option-#-type' (eg: option-1-text, option-2-image, option-3-correct...)
+      if (attr.name.match(/^option-\d+-\w+$/)) {
+        const splitParts = attr.name.split('-');
+        const optionNumber = parseInt(splitParts[1], 10);
+        while (options.length < optionNumber) {
+          options.push({'optionIndex': options.length});
+        }
```

Comment:
Are you trying to guarantee that the options are pushed in the array in the correct order? If yes, please add a short inline comment

Reply:
Yes, thats the idea